tests/backend/test_kv_store.py

```diff
@@ -0,0 +1,156 @@
+        client = MagicMock()
+        client().graphql.return_value = GraphQLResult(data=dict(key_value=[]))
+        monkeypatch.setattr("prefect.backend.kv_store.Client", client)
+
+        with pytest.raises(ValueError):
+            get_key_value(key="foo")
+
+        client.return_value.graphql.assert_called_with(
+            {"query": {'key_value(where: { key: { _eq: "foo" } })': {"value"}}}
+        )
+
+
+class TestDeleteKeyValue:
+    def test_delete_key_value_raises_on_server_backend(self, server_api):
+        with pytest.raises(ClientError):
+            delete_key(key="foo")
+
+    def test_get_key_value_calls_client_query_correctly(self, monkeypatch, cloud_api):
+        key_value_id_gql_return = GraphQLResult(
+            data=dict(
+                key_value=[GraphQLResult({"id": "123"})],
+            )
+        )
+        delete_key_value_gql_return = GraphQLResult(
+            data=dict(
+                delete_key_value=GraphQLResult({"success": True}),
+            )
+        )
+
+        # helper function to return key value id
+        # and the delete_key_value depending on input
+        def fake_graphql_responses(*args, **kwargs):
+            if "query" in kwargs["query"]:
+                return key_value_id_gql_return
+            return delete_key_value_gql_return
+
+        client = MagicMock()
+        client.return_value.graphql.side_effect = fake_graphql_responses
+        monkeypatch.setattr("prefect.backend.kv_store.Client", client)
+
+        success = delete_key(key="foo")
+
+        assert success
+
+        assert client.return_value.graphql.call_args_list == [
+            call(
+                query={"query": {'key_value(where: { key: { _eq: "foo" } })': {"id"}}}
+            ),
+            call(
+                query={
+                    "mutation($input: delete_key_value_input!)": {
+                        "delete_key_value(input: $input)": {"success"}
+                    }
+                },
+                variables={"input": {"key_value_id": "123"}},
+            ),
+        ]
+
+    def test_delete_key_value_raises_if_key_not_found(self, monkeypatch, cloud_api):
+        client = MagicMock()
+        client().graphql.return_value = GraphQLResult(
+            data=dict(
+                key_value=[],
+            )
+        )
+        monkeypatch.setattr("prefect.backend.kv_store.Client", client)
+
+        with pytest.raises(ValueError):
+            delete_key(key="foo")
+
+        client.return_value.graphql.assert_called_with(
+            query={"query": {'key_value(where: { key: { _eq: "foo" } })': {"id"}}}
+        )
+
+
+class TestListKeyValue:
+    def test_list_key_value_raises_on_server_backend(self, server_api):
+        with pytest.raises(ClientError):
+            list_keys()
+
+    def test_list_key_value_calls_client_mutation_correctly(
+        self, monkeypatch, cloud_api
+    ):
+        client = MagicMock()
+        client().graphql.return_value = GraphQLResult(
```

Comment:
Note that I capitalized this intentionally so it mirrors the `Client` you are mocking. Lowercase makes it less clear (imo) that you are mocking the class not the instance. Not a big deal though.

Reply:
fair point, updated